docs/src/main/asciidoc/databases-dev-services.adoc

```diff
@@ -61,6 +61,69 @@ ibmcom/db2:11.5.0.0a
 mcr.microsoft.com/mssql/server:2022-latest
 ----
 
+[[reuse]]
+== Reusing Dev Services
+
+[[reuse-within-execution]]
+=== General case
+
+Within a dev mode session or test suite execution,
+Quarkus will always reuse database Dev Services as long as their configuration
+(username, password, environment, port bindings, ...) did not change.
+
+When the configuration of any database Dev Services changes,
+Quarkus will always restart all database Dev Services.
+
+When a dev mode session or test suite execution ends,
+Quarkus will (by default) stop all database Dev Services.
+
+[[reuse-across-executions]]
+=== Reusing Dev Service containers across runs
+
+Assuming you rely on Dev Services based on containers (unlike H2 or Derby),
+if you want to keep Dev Service containers running *after a dev mode session or test suite execution*
+to reuse them in the next dev mode session or test suite execution,
+this is possible as well.
+Just enable https://java.testcontainers.org/features/reuse/[TestContainers reuse]
+by inserting this line in one of your
+https://java.testcontainers.org/features/configuration/[TestContainers configuration file]
+(generally `~/.testcontainers.properties` or `C:/Users/myuser/.testcontainers.properties`):
+
+[source,properties]
+----
+testcontainers.reuse.enable=true
+----
+
+[NOTE]
+====
+Even with container reuse enabled, containers will only be reused if their startup command did not change:
+same environment variables (username/password in particular), same port bindings, same volume mounts, ...
+====
+
+[WARNING]
+====
+Reusing containers implies reusing their internal state,
+including the database schema and the content of tables.
+
+If that's not what you want -- and if your tests write to the database, that's probably not what you want --
+consider xref:hibernate-orm.adoc#dev-mode[configuring Hibernate ORM appropriately],
+or using xref:flyway.adoc[Flyway] or xref:liquibase.adoc[Liquibase].
+====
+
+[WARNING]
+====
+With container reuse enabled, old containers (especially with obsolete configuration)
+might be left running indefinitely, even after starting a new Quarkus dev mode session or test suite execution.
+
+In that case, you will need to stop and remove these containers manually.
+====
+
+If you want to reuse containers for some Quarkus applications but not all of them,
+or some Dev Services but not all of them,
+you can disable this feature for a specific Dev Service by setting the configuration property
+xref:databases-dev-services.adoc#quarkus-datasource-config-group-dev-services-build-time-config_quarkus.datasource.devservices.reuse[`quarkus.datasource.devservices.reuse`]
+to `false`.
```

Comment:
Should we add something about named datasources?

Reply:
I'll do it, but it definitely makes things more convoluted, and we don't do it consistently in this guide (and probably others)...
We probably need a macro somewhere :D